plugins/tiddlywiki/multiwikiserver/modules/store/sql-tiddler-store.js

```diff
@@ -151,8 +151,9 @@ SqlTiddlerStore.prototype.processIncomingTiddler = function(tiddlerFields) {
 	const attachmentsEnabled = this.adminWiki.getTiddlerText("$:/config/MultiWikiServer/EnableAttachments","yes") === "yes";
 	const contentTypeInfo = $tw.config.contentTypeInfo[tiddlerFields.type || "text/vnd.tiddlywiki"],
 		isBinary = !!contentTypeInfo && contentTypeInfo.encoding === "base64";
-	if(attachmentsEnabled && isBinary && tiddlerFields.text && tiddlerFields.text.length > attachmentSizeLimit) {
-		const attachment_blob = this.attachmentStore.saveAttachment({
+
+	if (attachmentsEnabled && isBinary && ((tiddlerFields.text && tiddlerFields.text.length > attachmentSizeLimit) || (!!tiddlerFields?.attachment_blob && tiddlerFields?.attachment_blob?.length < attachmentSizeLimit))) {
+		const attachment_blob = tiddlerFields?.attachment_blob ?? this.attachmentStore.saveAttachment({
```

Comment:
Another unusual thing about TiddlyWiki is that in the core we target quite old fashioned JavaScript, aiming to hit compatibility with ECMAScript 5.1. That's why we don't use classes etc. The motivation is to allow TiddlyWiki to run on as wide a range of browsers as possible.

MWS is running in a slightly different situation, where there are very few good reasons for people to be running old versions of Node. So, the plan is to formally adopt a later dialect of JS within MWS. We haven't settled on a version, but I am guessing it might be Node.js 18.

That means that I think it's OK to use the optional chaining operator here, but we should be careful of adopting anything much more recent than that.